src/system-health/health_checker/service_checker.py

```diff
@@ -24,23 +43,177 @@ class ServiceChecker(HealthChecker):
+            feature_table (object): FEATURE table in CONFIG_DB
+
+        Returns:
+            expected_running_containers: A set of container names that are expected running
+            container_feature_dict: A dictionary {<container_name>:<feature_name>}
+        """
+        expected_running_containers = set()
+        container_feature_dict = {}
+        for feature_name, feature_entry in feature_table.items():
+            if feature_entry["state"] not in ["disabled", "always_disabled"]:
+                if multi_asic.is_multi_asic():
+                    if feature_entry["has_global_scope"] == "True":
+                        expected_running_containers.add(feature_name)
+                        container_feature_dict[feature_name] = feature_name
+                    if feature_entry["has_per_asic_scope"] == "True":
+                        num_asics = multi_asic.get_num_asics()
+                        for asic_id in range(num_asics):
+                            expected_running_containers.add(feature_name + str(asic_id))
+                            container_feature_dict[feature_name + str(asic_id)] = feature_name
+                else:
+                    expected_running_containers.add(feature_name)
+                    container_feature_dict[feature_name] = feature_name
+
+        return expected_running_containers, container_feature_dict
+
+    def get_current_running_containers(self):
+        """Get current running containers, if the running container is not in self.container_critical_processes,
+           try get the critical process list
+
+        Returns:
+            running_containers: A set of running container names
+        """
+        DOCKER_CLIENT = docker.DockerClient(base_url='unix://var/run/docker.sock')
+        running_containers = set()
+        ctrs = DOCKER_CLIENT.containers
+        try:
+            lst = ctrs.list(filters={"status": "running"})
+
+            for ctr in lst:
+                running_containers.add(ctr.name)
+                if ctr.name not in self.container_critical_processes:
+                    self.fill_critical_process_by_container(ctr.name)
+        except docker.errors.APIError as err:
+            logger.log_debug("Failed to retrieve the running container list. Error: '{}'".format(err))
+
+        return running_containers
+
+    def get_critical_process_list_from_file(self, container, critical_processes_file):
+        """Read critical process name list from critical processes file
+
+        Args:
+            container (str): contianer name
+            critical_processes_file (str): critical processes file path
+
+        Returns:
+            critical_process_list: A list of critical process names
+        """
+        critical_process_list = []
+
+        with open(critical_processes_file, 'r') as file:
+            for line in file:
+                # Try to match a line like "program:<process_name>"
+                match = re.match(r"^\s*((.+):(.*))*\s*$", line)
```

Comment:
For this regex matching, if the content of variable `line` is `      program   :    lldpd   `, then the `identifier_key` and `identifier_value` will have the preceeding and trailing white spaces, do we need remove them?